pipelines/azure-pipelines.publish.yml

```diff
@@ -0,0 +1,153 @@
+          - task: SFP.build-tasks.custom-build-task-1.EsrpCodeSigning@5
+            displayName: 'Sign manifest'
+            inputs:
+              ConnectedServiceName: AppInstallerESRPCodeSigning
+              AppRegistrationClientId: '32216f16-efc9-4013-9fae-c6a2c54a3fc0'
+              AppRegistrationTenantId: '72f988bf-86f1-41af-91ab-2d7cd011db47'
+              AuthAKVName: PeetDevOpsKeyVault
+              AuthCertName: ESRPAuth
+              AuthSignCertName: ESRPRequestSigning
+              FolderPath: '$(System.DefaultWorkingDirectory)\ToSign\'
+              Pattern: '*'
+              signConfigType: inlineSignParams
+              inlineOperation: |
+                          [
+                            {
+                                "KeyCode" : "CP-230012",
+                                "OperationCode" : "SigntoolSign",
+                                "Parameters" : {
+                                    "OpusName" : "Microsoft",
+                                    "OpusInfo" : "http://www.microsoft.com",
+                                    "FileDigest" : "/fd \"SHA256\"",
+                                    "PageHash" : "/NPH",
+                                    "TimeStamp" : "/tr \"http://rfc3161.gtm.corp.microsoft.com/TSS/HttpTspServer\" /td sha256"
+                                },
+                                "ToolName" : "sign",
+                                "ToolVersion" : "1.0"
+                            },
+                            {
+                                "KeyCode" : "CP-230012",
+                                "OperationCode" : "SigntoolVerify",
+                                "Parameters" : {},
+                                "ToolName" : "sign",
+                                "ToolVersion" : "1.0"
+                            }
+                          ]
+          - task: PowerShell@2
+            displayName: Copy Signed Files and Validate signature
+            inputs:
+              targetType: inline
+              pwsh: true
+              script: |
+                New-Item ToPublish\${{ parameters.moduleName }} -Type Directory
+                $moduleFolder = "ToPublish\${{ parameters.moduleName }}"
+                Copy-Item -Path "ToSign\${{ parameters.moduleName }}.psm1" -Destination "ToPublish\${{ parameters.moduleName }}\${{ parameters.moduleName }}.psm1" -Force
+                Copy-Item -Path "ToSign\${{ parameters.moduleName }}.psd1" -Destination "ToPublish\${{ parameters.moduleName }}\${{ parameters.moduleName }}.psd1" -Force
+                
+                $notValid = Get-ChildItem $moduleFolder -Recurse -Attributes !Directory | Get-AuthenticodeSignature | where { $_.Status -ne 'Valid' }
+                if ($null -ne $notValid)
+                {
+                    $notValid
+                    throw "A file is not signed"
```

Comment:
nit: is it possible to get which file is not signed?

Reply:
$notValid should list out the items that are not signed.